src/coreclr/src/pal/src/misc/utils.cpp

```diff
@@ -332,6 +332,9 @@ void UTIL_SetLastErrorFromMach(kern_return_t MachReturn)
 --*/
 BOOL IsRunningOnMojaveHardenedRuntime()
 {
+#if defined(TARGET_ARM64)
+    return true;
```

Comment:
I am not sure what motivated the change originally.  It was probably early in draft W^X implementation.  I might have put it in the optimization when I was reading through the code (partially because I wouldn't have to debug it).  It looks to me like it should work correctly on Apple Silicon and on x64 Big-Sur.

Any objections to leaving this as an optimization?

FWIW, the name of this function will eventually be arcane.  Mojave will be a distant memory...  Big-Sur has Mojave Hardening enable, but semantically `isRunningOnMojaveHardenedRuntime` should be false...  While `requiresMapJitFlag` may be less confusing.